src/app/components/store-products/store-products.component.ts

```diff
@@ -8,4 +19,39 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class StoreProductsComponent {
+  readonly storeId$: Observable<string> = this._activatedRoute.params.pipe(
+    map((params) => params['storeId']),
+    shareReplay(1)
+  );
+  readonly tagsMap$: Observable<Record<string, TagModel>> = this._tagsService.getTagsMap();
+
+  readonly storeDetails$: Observable<StoreQueryModel> = combineLatest([this.storeId$, this.tagsMap$]).pipe(
+    switchMap(([storeId, tagsMap]) =>
+      this._storesService.getSingleStoreById(storeId).pipe(map((store) => this.mapToStoreQueryModel(tagsMap, store)))
+    )
+  );
+
+  readonly storeSearch: FormGroup = new FormGroup({ keyWord: new FormControl() });
+
+  constructor(
+    private _activatedRoute: ActivatedRoute,
+    private _storesService: StoresService,
+    private _productsService: ProductsService,
+    private _tagsService: TagsService
+  ) {}
+
+  mapToStoreQueryModel(tagsMap: Record<string, TagModel>, store: StoreModel): StoreQueryModel {
+    return {
+      id: store.id,
+      name: store.name,
+      logoUrl: store.logoUrl,
+      distanceInKm: Math.round(store.distanceInMeters / 100) / 10,
+      tags: store.tagIds.map((tagId) => tagsMap[tagId]?.name)
+    };
+  }
+
+  onStoreSearchSubmitted(storeSearch: FormGroup): void {
```

Comment:
we should probably remove it

Reply:
This function will be used later so I only removed the body that I used for debugging.